iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -789,6 +754,14 @@ private async Task OpenAsyncInternalAsync(CancellationToken cancellationToken)
             {
                 try
                 {
+                    if (_channel != null
+                        && _channel is IDisposable disposableChannel)
+                    {
+                        if (Logging.IsEnabled)
+                            Logging.Info(this, $"_channel is disposable; disposing", nameof(OpenInternalAsync));
+                        disposableChannel.Dispose();
+                        _channel = null;
```

Comment:
q - I don't think this is needed here, since with state as `NotInitialized`, the channel will always be null. Is this extra level of cleanup required?

Reply:
It isn't strictly required in this case, but I find it to be good to return to a known state, and it doesn't cost much to do that.